### PR TITLE
bpo-33509: Fix _warnings for module_globals=None

### PR DESCRIPTION
**Pull request by vstinner**
Don't crash on warnings.warn_explicit() if module_globals is None.

https://bugs.python.org/issue33509

### CODE REVIEW

**Review by 1st1 — APPROVED** [review]

File: Python/_warnings.c [hunk 1]
```diff
@@ -883,6 +883,13 @@ get_source_line(PyObject *module_globals, int lineno)
     PyObject *source_list;
     PyObject *source_line;
 
+    if (!PyDict_Check(module_globals)) {
```
Comment (serhiy-storchaka) [inline Python/_warnings.c]: I would move this check into `warnings_warn_explicit()`.

File: Python/_warnings.c [hunk 2]
```diff
@@ -883,6 +883,13 @@ get_source_line(PyObject *module_globals, int lineno)
...
+        PyErr_Format(PyExc_TypeError,
+                     "module_globals must be a dict, " "not '%.200s'",
```
Comment (serhiy-storchaka) [inline Python/_warnings.c]: Why `" "`?

Comment (vstinner) [reply Python/_warnings.c]: fixed

File: Python/_warnings.c (hunk 1 above)
Comment (vstinner) [reply Python/_warnings.c]: fixed

**Comment by serhiy-storchaka** [comment]
Sorry, I don't see you latest changes Victor.

**Comment by vstinner** [comment]
> Sorry, I don't see you latest changes Victor.

Woops. I just forgot to commit my changes... They are now published.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**